settings_vial/settings_vial.py

```diff
@@ -0,0 +1,110 @@
+import json
+import warnings
+from os import environ
+
+from .exceptions import MissingOverrideKeys, NotCallable, UnsupportedSetType
+
+
+class Settings:
+    r"""Settings from environment variables.
+
+    :param env_prefix: prefix to be used when looking through the environment for variables
+    :param override_prefix: prefix to be used when looking for overrideable variables
+    :param override_callable: callable to be used when retrieving the set of keys to be overriden
+
+    Usage::
+
+        $ export MY_APP_TEST_VAR=42
+
+        >>> from settings_vial import Settings
+        >>> settings = Settings(env_prefix="MY_APP_")
+        >>> settings.load_env()
+        >>> settings.TEST_VAR
+        42
+    """
+
+    def __init__(self, env_prefix, override_prefix=None, override_callable=None):
```

Comment:
Not certain about the actual use of override_callable; I see the tests and that they pass.

Reply:
So, think of the following scenario, I have a certain setting for enabling throttling in certain endpoints, by default it's turned off but in my e2e tests I want to enable it, but only for a certain set of my tests/requests.
Passing a callable (e.g. a function that fetches specific request headers) that returns the key(s) you defined as your override settings, it will use those alternative settings for that request. It is mainly aimed at e2e tests.

Reply:
I think `override_callable` is not a good name. What about `override_keys_enumerator` or `override_keys_function`?